### DISCUSSION TITLE
LibreChat Apps (iOS, Android, Mac, Windows)

### DISCUSSION THREAD

**Discussion by ClaraLeigh** [danny-avila/LibreChat#695] [General]
Just starting a thread to talk about the idea of having navive apps of the current librachat app. 
I'm thinking the stack would be react native + nodejs.

With the concept being:
1. an initial setup for the app only, allowing you to essentially edit what is currently the .env file. So you could input your own mongodb, api keys etc
2. Run everything in the app, so no external servers needed. You can run the nodejs inside it and have everything self contained. No hosting required.
3. Split some of the frontend code out into react components so it can be used easier in react native

Note: I would have added this as a discussion but idk if that works with github's "Projects"

**Comment by lsalazarm99** [comment]
I didn't understand the point 2 🤔. It sounds to me that you would like the mobile app to be able to work without the LibreChat backend.

While it's possible, why would you do that if you already have the server with all its functionalities? It sounds like a lot of double work.

**Comment by githubdebugger** [comment]
This https://github.com/matthaigh27/ChatGPT-android-app is a good webview app for ChatGPT, I am not an Android dev, but I guess changing the URL to the server where LibreChat is hosted might do the trick and some additional stuffs?

**Reply by eburnette** [reply]
That wouldn't do it, you'd need to port the client side of librechat to a standalone application.

**Reply by fuegovic** [reply]
someone did add support for LibreChat to this app, you can check out his fork here:
https://github.com/goodair220917/LibreChat-Android-App

**Comment by leikoilja** [comment]
I have just open-sourced a desktop app wrapper called LibreChat-UI [here](https://github.com/leikoilja/LibreChat-UI) if you are interested. I've been using it for the past 3 weeks and it has been working great for me :)

**Comment by dcolley** [comment]
What about PWA, just adding the web page to Home?
Works fine on iOS for me.

PWA is fine for a client.
_(Unless you mean to port the backend as well?)_

**Reply by jamiehannaford** [reply]
I think this only works for Safari on iOS. AFAIK browsers like Brave, Chrome, Firefox don't support PWA

**Reply by PetbkA** [reply]
1. Chrome [does](https://support.google.com/chrome/answer/9658361) support PWA;
2. Firefox doesn't but it has [extension](https://addons.mozilla.org/en-US/firefox/addon/pwas-for-firefox/) for this, although it's not PWA from performance view 'cause it runs in another "instance" of browser instead of working as tab process of current (may be something changed in current versions);
3. Some FF mods like [Floorp](https://floorp.app/en-US) support PWA which even works in current instance of browser

**Comment by JumpLink** [comment]
What are your thoughts on native apps that connect to the server version instead? Kind of like the mobile version of ChatGPT, but for LibreChat? That would probably require something like an Backend API and a way to generate tokens for such clients

**Comment by vinicius507** [comment]
A client app for mobile would be nice. Specially if it could be integrated into device assistant APIs. What are your thoughts on this?

**Comment by garfiec** [comment]
@vinicius507 I've also been waiting a while for a native mobile app of LibreChat. Eventually just built my own because the PWA wasn't the experience I wanted. Connects to your existing server (v0.8.4) and covers the core features that I personally use

Here's the repo if anyone wants to take a look: https://github.com/garfiec/Librechat-Mobile

**Reply by F4bC0d3** [reply]
Can u pls add built apks in releases.

**Reply by ifedan-ed** [reply]
https://cloud.danvics.com/s/BkW8wZQHJq9Pqi2

> Can u pls add built apks in releases.

https://cloud.danvics.com/s/BkW8wZQHJq9Pqi2

**Reply by garfiec** [reply]
@F4bC0d3 @ifedan-ed 

Releases are now added. You can also use Obtainium to track new releases.

https://github.com/garfiec/Librechat-Mobile#obtainium-recommended

**Reply by altinturk** [reply]
@garfiec kudos. Amazing result, looks great, works nice
